litestar/channels/plugin.py

```diff
@@ -0,0 +1,431 @@
+from __future__ import annotations
+
+import asyncio
+from asyncio import CancelledError, Queue, QueueFull, Task, create_task
+from collections import deque
+from contextlib import AsyncExitStack, asynccontextmanager, suppress
+from functools import partial
+from os.path import join as join_path
+from typing import TYPE_CHECKING, AsyncGenerator, Awaitable, Callable, Generic, Iterable, Literal, Sequence, TypeVar
+
+import anyio
+import msgspec.json
+
+from litestar.di import Provide
+from litestar.exceptions import ImproperlyConfiguredException, LitestarException
+from litestar.handlers import WebsocketRouteHandler
+from litestar.plugins import InitPluginProtocol
+from litestar.serialization import default_serializer
+
+if TYPE_CHECKING:
+    from litestar.channels.base import ChannelsBackend
+    from litestar.config.app import AppConfig
+    from litestar.connection import WebSocket
+    from litestar.types import LitestarEncodableType, TypeEncodersMap
+    from litestar.types.asgi_types import WebSocketMode
+
+
+T = TypeVar("T")
+
+
+class ChannelsException(LitestarException):
+    pass
+
+
+class AsyncDeque(Queue, Generic[T]):
```

Comment:
What makes this Async?

Reply:
It subclasses `Queue`, which is an async queue.